test/gtest/conv_2d_implicit_gemm.cpp

```diff
@@ -0,0 +1,241 @@
+class Conv2dFloat : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+void Run2dDriver(Precision prec)
+{
+
+    std::vector<TestCase> params;
+    switch(prec)
+    {
+    case Precision::Float: params = Conv2dFloat::GetParam(); break;
+    case Precision::Half: params = Conv2dHalf::GetParam(); break;
+    case Precision::Int8: params = Conv2dInt8::GetParam(); break;
+    case Precision::BFloat16: params = Conv2dBFloat16::GetParam(); break;
+    default: params = Conv2dFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        for(std::string const& str : tokens)
+            ptrs.push_back(str.data());
+
+        testing::internal::CaptureStderr();
+        test_drive<conv2d_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        EXPECT_FALSE(capture.find("Perf Db: record not found") != std::string::npos);
+    }
+};
+
+TEST_P(Conv2dFloat, FloatTest)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if(!miopen::StartsWith(handle.GetDeviceName(), "gfx906") || GetFloatArg() != "--float")
+    {
+        GTEST_SKIP();
+    }
+    else
+    {
+        Run2dDriver(Precision::Float);
+    }
+
+#else
+    GTEST_SKIP();
+#endif
+};
+
+TEST_P(Conv2dHalf, HalfTest)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if(!miopen::StartsWith(handle.GetDeviceName(), "gfx906") || GetFloatArg() != "--half")
+    {
+        GTEST_SKIP();
+    }
+    else
+    {
+        Run2dDriver(Precision::Half);
+    }
+
+#else
+    GTEST_SKIP();
+#endif
+};
+
+TEST_P(Conv2dInt8, Int8Test)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if(!miopen::StartsWith(handle.GetDeviceName(), "gfx906") || GetFloatArg() != "--int8")
+    {
+        GTEST_SKIP();
+    }
+    else
+    {
+        Run2dDriver(Precision::Int8);
+    }
+
+#else
+    GTEST_SKIP();
+#endif
+};
+
+TEST_P(Conv2dBFloat16, BFloat16Test)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if(!miopen::StartsWith(handle.GetDeviceName(), "gfx906") || GetFloatArg() != "--bfloat16")
+    {
+        GTEST_SKIP();
+    }
+    else
+    {
+        Run2dDriver(Precision::BFloat16);
+    }
+
+#else
+    GTEST_SKIP();
+#endif
+};
+
+std::vector<TestCase> GetTestCases(const std::string& precision)
+{
+    const std::vector<TestCase> test_cases = {
+        // clang-format off
+    std::make_tuple<std::vector<std::string>, std::string>(
+        {"MIOPEN_DEBUG_AMD_WINOGRAD_RXS_F3X2=0", "MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_WRW_V4R1=0"},
+        precision + " --disable-validation --verbose --input 128 1024 14 14 --weights 2048 1024 1 1 --pads_strides_dilations 0 0 2 2 1 1"),
+    std::make_tuple<std::vector<std::string>, std::string>(
+        {"MIOPEN_DEBUG_AMD_WINOGRAD_RXS_F3X2=0", "MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_FWD_V4R1=0"},
```

Comment:
@alexandraBara Also you can split lines manually by means of EOL comments, and keep formatting enabled:
```C++
  some text //
    you want to split //
    manually;
```

Reply:
> The env vars need to be factored out into constants to reduce the amount of lines, since there are only three unique combinations

@JehandadKhan , shortened the env vars